### PR TITLE
fix: support chart on bsx collection

### LINKED ISSUE #3646: Fix charts on Basilisk in collection
Time to fix the charts on collections in Basilisk. 
We are not using current Basilisk data in charts. 

Add floor prices and buy events as we have on RMRK charts.
 
https://beta.kodadot.xyz/bsx/collection/3612943633?tab=chart
image

image

test cases with many buys in collection
- https://beta.kodadot.xyz/bsx/collection/2645998698?tab=history&event=BUY
- https://beta.kodadot.xyz/bsx/collection/515729876?tab=history&event=BUY
- https://beta.kodadot.xyz/bsx/collection/2600576003?tab=history&event=BUY
- https://beta.kodadot.xyz/bsx/collection/2425975522?tab=history&event=BUY
- https://beta.kodadot.xyz/bsx/collection/1679439669?tab=history

### PR DESCRIPTION
**Pull request by preschian**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring

### What's new?

- [x] PR closes #3646 
  - move collectionPriceChart to shared components
  - refactor line chart

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged the recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

### Optional

- [x] I've tested it at </rmrk/collection/2644199cf3652aaa78-KK01?tab=chart>
  - https://deploy-preview-3651--koda-nuxt.netlify.app/bsx/collection/3612943633?tab=chart
  - https://deploy-preview-3651--koda-nuxt.netlify.app/bsx/collection/2645998698?tab=chart
  - https://deploy-preview-3651--koda-nuxt.netlify.app/bsx/collection/515729876?tab=chart
  - https://deploy-preview-3651--koda-nuxt.netlify.app/bsx/collection/2600576003?tab=chart
  - https://deploy-preview-3651--koda-nuxt.netlify.app/bsx/collection/2425975522?tab=chart
  - https://deploy-preview-3651--koda-nuxt.netlify.app/bsx/collection/1679439669?tab=chart
- [ ] I've tested PR on mobile and everything seems works
- [ ] I found edge cases
- [ ] I've written some unit tests 🧪

### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=DY4SQF2iD456tH89aQtz5wv1EV3BbSW8wKKuMcwbmXaj1pM)

### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot

- [ ] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

### CODE REVIEW

**Comment by yangwao** [comment]
https://deploy-preview-3651--koda-nuxt.netlify.app/bsx/collection/3248880134?tab=chart

seems some buys happened here, but don't see them on chart? 

image


image

**Comment by preschian** [comment]
currently, I group them by the same date and combine the value as average

this is what the chart looks like without grouping the data. should I change with this instead?
URL: https://deploy-preview-3651--koda-nuxt.netlify.app/bsx/collection/515729876?tab=chart&page=1
Screen Shot 2022-08-03 at 16 52 05

**Comment by petersopko** [comment]
@kkukelka @cryptodamsky can we review this one?

File: components/bsx/Collection/ChartContainer.vue [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      variables: {
+        id: this.id,
+        and: {
+          interaction_eq,
+        },
+      },
+    })
+  }
+
+  protected formatValue(value) {
```
Comment (kkukelka) [inline components/bsx/Collection/ChartContainer.vue:49]: please add function parameter type and return type

File: components/shared/chart/chartMixin.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+export default class ChartMixin extends Vue {
+  @Prop() labels!: string[]
+  @Prop() datasets!: ChartDataset[]
+  @Prop() options
+
+  protected chart!: Chart
+  protected type: keyof ChartTypeRegistry = 'line'
+
+  protected renderChart() {
+    const ctx = (this.$refs.canvas as HTMLCanvasElement).getContext('2d')
```
Comment (kkukelka) [inline components/shared/chart/chartMixin.ts:22]:
```suggestion
    const ctx = (this.$refs.canvas as HTMLCanvasElement)?.getContext('2d')
```

File: components/shared/collection/PriceChart.vue [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+              color: '#3a3a3a',
+            },
+          },
+        },
+      }
+    }
+  }
+
+  @Watch('priceData')
+  async watchData() {
```
Comment (kkukelka) [inline components/shared/collection/PriceChart.vue:170]: async but no await on call? is `priceChart()` even a asynchronous function?

**Review by kkukelka — CHANGES_REQUESTED** [review]

File: components/shared/collection/PriceChart.vue (hunk 3 above)
Comment (preschian) [reply components/shared/collection/PriceChart.vue]: oh you're right, updated 👍

**Review by kkukelka — APPROVED** [review]
lg now

**Review by cryptodamsky — APPROVED** [review]

**Comment by kkukelka** [comment]
pay 100

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $100 @ 62.74 USD/KSM ~ 1.594 $KSM
🧗 [DY4SQF2iD456tH89aQtz5wv1EV3BbSW8wKKuMcwbmXaj1pM](https://kusama.subscan.io/account/DY4SQF2iD456tH89aQtz5wv1EV3BbSW8wKKuMcwbmXaj1pM)
🔗 [0x38dcf6c16d37e6c6af1d82bba2b43a6cf7b0cd7ef1d3b7958510f765d4a2c0a9](https://kusama.subscan.io/extrinsic/0x38dcf6c16d37e6c6af1d82bba2b43a6cf7b0cd7ef1d3b7958510f765d4a2c0a9)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**